code/modules/medical_machinery/machines/advanced_medical_machinery.dm

```diff
@@ -0,0 +1,83 @@
+/obj/machinery/medical/dialysis
+	name = "Direct Blood Filtration Unit"
+	desc = "Automatically filtrates your blood from all chemicals."
+	icon_state = "dialysis"
+	///Amount of purged chems per process
+	var/purge_amount = 3
+
+/obj/machinery/medical/dialysis/RefreshParts()
+	for(var/obj/item/stock_parts/capacitor/C in component_parts)
+		purge_amount = initial(purge_amount) * C.rating
```

Comment:
This does not properly support multiple of the same part. You should follow what other machines do.

Reply:
fixed